### PR TITLE
Split ReportPoint into histograms and metrics

### PR DESCRIPTION
**Pull request by basilisk487**

### CODE REVIEW

File: java-lib/src/main/avro/Reporting.avdl [hunk 1]
```diff
@@ -110,4 +110,22 @@ protocol Reporting {
     union {null, array<string>} tags = null;
     union {null, map<array<string>>} dimensions = null;
   }
+
+  record ReportMetric {
+    string metric;
+    long timestamp;
+    double value;
+    string host = "unknown";
+    string table = "tsdb";
```
Comment (panghy) [inline java-lib/src/main/avro/Reporting.avdl:119]: let's rename this to customer

File: java-lib/src/main/avro/Reporting.avdl [hunk 2]
```diff
@@ -110,4 +110,22 @@ protocol Reporting {
...
+    map<string> annotations = {};
+  }
+
+  record ReportHistogram {
+    string metric;
+    long timestamp;
+    Histogram value;
+    string host = "unknown";
+    string table = "tsdb";
```
Comment (panghy) [inline java-lib/src/main/avro/Reporting.avdl:128]: ditto

File: java-lib/src/main/avro/Reporting.avdl [hunk 3]
```diff
@@ -110,4 +110,22 @@ protocol Reporting {
...
     union {null, map<array<string>>} dimensions = null;
   }
+
+  record ReportMetric {
+    string metric;
+    long timestamp;
+    double value;
+    string host = "unknown";
+    string table = "tsdb";
+    map<string> annotations = {};
```
Comment (panghy) [inline java-lib/src/main/avro/Reporting.avdl:120]: Ultimately to support multi-valued keys, we need to use a class unfortunately, this is the chance to pull it off

File: java-lib/src/main/avro/Reporting.avdl [hunk 4]
```diff
@@ -110,4 +110,22 @@ protocol Reporting {
...
+  }
+
+  record ReportHistogram {
+    string metric;
+    long timestamp;
+    Histogram value;
+    string host = "unknown";
+    string table = "tsdb";
+    map<string> annotations = {};
```
Comment (panghy) [inline java-lib/src/main/avro/Reporting.avdl:129]: ditto

File: java-lib/src/main/avro/Reporting.avdl (hunk 3 above)
Comment (basilisk487) [reply java-lib/src/main/avro/Reporting.avdl]: done

File: java-lib/src/main/java/com/wavefront/data/AnnotationUtils.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    if (annotations == null) return null;
+    final ImmutableSetMultimap.Builder<String, String> builder = ImmutableSetMultimap.builder();
+    for (Annotation annotation : annotations) {
+      builder.put(annotation.getKey(), annotation.getValue());
+    }
+    return builder.build();
+  }
+
+  @Nullable
+  public static List<Annotation> toAnnotationList(Multimap<String, String> tags) {
```
Comment (panghy) [inline java-lib/src/main/java/com/wavefront/data/AnnotationUtils.java:35]: mark @Nullable for tags

File: java-lib/src/main/java/com/wavefront/data/AnnotationUtils.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+ *
+ * @author vasily@wavefront.com
+ */
+public abstract class AnnotationUtils {
+  private AnnotationUtils() {
+  }
+
+  @Nullable
+  public static SetMultimap<String, String> toMultimap(@Nullable Collection<Annotation> annotations) {
+    if (annotations == null) return null;
```
Comment (panghy) [inline java-lib/src/main/java/com/wavefront/data/AnnotationUtils.java:26]: sometimes I find that returning empty multimaps to be easier to reason (for callee)

File: java-lib/src/main/java/com/wavefront/data/AnnotationUtils.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    final ImmutableSetMultimap.Builder<String, String> builder = ImmutableSetMultimap.builder();
+    for (Annotation annotation : annotations) {
+      builder.put(annotation.getKey(), annotation.getValue());
+    }
+    return builder.build();
+  }
+
+  @Nullable
+  public static List<Annotation> toAnnotationList(Multimap<String, String> tags) {
+    if (tags == null || tags.isEmpty()) return null;
```
Comment (panghy) [inline java-lib/src/main/java/com/wavefront/data/AnnotationUtils.java:36]: ditto

**Pull request merged**